specs/experimental/inbox-contract.md

```diff
@@ -0,0 +1,315 @@
+    uint32 _blobbasefeeScalar,
+    bytes32 _batcherHash,
+    uint64 _gasLimit,
+    address _unsafeBlockSigner,
+    ResourceMetering.ResourceConfig memory _config,
+    address _batchInbox,
+    SystemConfig.Addresses memory _addresses
+)
+    public
+    initializer
+{
+    ...
+    // Storage.setAddress(BATCH_INBOX_SLOT, _batchInbox);
+    // initialize inbox by `_setBatchInbox` so that an event is emitted.
+    _setBatchInbox(_batchInbox);
+    ...
+}
+```
+
+#### UpdateType
+
+A new enum value `BATCH_INBOX` is added to the `UpdateType` enumeration.
+
+```solidity
+enum UpdateType {
+    BATCHER,
+    GAS_CONFIG,
+    GAS_LIMIT,
+    UNSAFE_BLOCK_SIGNER,
+    BATCH_INBOX
+}
+```
+
+### L1 Info Deposit Transaction
+
+The [`L1InfoDeposit`](https://github.com/ethereum-optimism/optimism/blob/5e317379fae65b76f5a6ee27581f0e62d2fe017a/op-node/rollup/derive/l1_block_info.go#L264) function creates a L1 Info deposit transaction based on the L1 block. It is extended to add an additional `batchInbox` parameter for the [`setL1BlockValuesEcotone`](https://github.com/ethereum-optimism/optimism/blob/5e317379fae65b76f5a6ee27581f0e62d2fe017a/packages/contracts-bedrock/src/L2/L1Block.sol#L136) function after the inbox contract feature is activated.
+
+```golang
+func L1InfoDeposit(rollupCfg *rollup.Config, sysCfg eth.SystemConfig, seqNumber uint64, block eth.BlockInfo, l2BlockTime uint64) (*types.DepositTx, error) {
+    ...
+    if isInboxForkButNotFirstBlock(rollupCfg, l2BlockTime) {
+        l1BlockInfo.BatchInbox = sysCfg.BatchInbox
+        l1BlockInfo.BlobBaseFee = block.BlobBaseFee()
+        if l1BlockInfo.BlobBaseFee == nil {
+          // The L2 spec states to use the MIN_BLOB_GASPRICE from EIP-4844 if not yet active on L1.
+          l1BlockInfo.BlobBaseFee = big.NewInt(1)
+        }
+        scalars, err := sysCfg.EcotoneScalars()
+        if err != nil {
+          return nil, err
+        }
+        l1BlockInfo.BlobBaseFeeScalar = scalars.BlobBaseFeeScalar
+        l1BlockInfo.BaseFeeScalar = scalars.BaseFeeScalar
+        // marshalBinaryInboxFork adds an additional `batchInbox` parameter based on marshalBinaryEcotone.
+        out, err := l1BlockInfo.marshalBinaryInboxFork()
+        if err != nil {
+          return nil, fmt.Errorf("failed to marshal InboxFork l1 block info: %w", err)
+        }
+        data = out
+    } else if isEcotoneButNotFirstBlock(rollupCfg, l2BlockTime) {
+    ...
+}
+```
+
+The `marshalBinaryInboxFork` function is added to [`L1BlockInfo`](https://github.com/ethereum-optimism/optimism/blob/5e317379fae65b76f5a6ee27581f0e62d2fe017a/op-node/rollup/derive/l1_block_info.go#L41). This new function incorporates an additional `batchInbox` parameter for the [`setL1BlockValuesEcotone`](https://github.com/ethereum-optimism/optimism/blob/5e317379fae65b76f5a6ee27581f0e62d2fe017a/packages/contracts-bedrock/src/L2/L1Block.sol#L136) function:
+
+```golang
+func (info *L1BlockInfo) marshalBinaryInboxFork() ([]byte, error) {
```

Comment:
We should define the spec in terms of the serialization instead of source code. Its not super clear exactly what this is doing without knowing the implementation details of various functions used

Reply:
This part is gone after following the new design pattern in https://github.com/ethereum-optimism/specs/pull/284/commits/f5b81785617bda2d22464585a39e1c2815394007.